scripts/qa_crowbarsetup.sh

```diff
@@ -3860,6 +3903,81 @@ function oncontroller_heat_image_setup()
     fi
 }
 
+function oncompute_parse_vm_xml()
+{
+    local instance_name=$1
+    echo $instance_name
+    virsh list --all | grep $instance_name
+    [ $? != 0 ] && complain 53 "failed to find pci-passthru-instance"
+
+    # verify node xml to see if it has pci passthru section which has the
+    # specific address passing from compute host
+    virsh dumpxml $instance_name | grep -E "<hostdev" -A 7 | grep "0x1d"
+    [ $? != 0 ] && complain 63 "failed to find pci-passthru block in node xml"
+
+    return 0
+}
+
+function oncontroller_setup_pci_passthru_vm()
+{
+    local cirros_image_url=$imageserver_url/$arch/openstack/cirros-0.4.0-x86_64-disk.img
```

Comment:
Unless I'm missing something, retrieving a cirros image is not specific to this passthru use case and therefore should be in a separate function for reusability.  (Even if it was specific, it would be more readable in a separate function anyway.)  So please can you extract that out?  Ditto for uploading to glance.